src/core/pagination/services.py

```diff
@@ -1,17 +1,21 @@
 from sqlalchemy import func, Table, select
 from sqlalchemy.orm import Session
 
-from src.core.pagination.models import BaseModel, PageParams, T, conint
-from src.core.pagination.schemas import PagedResponseSchema
+from src.core.pagination.models import BaseModel, PageParams, conint
+from src.core.pagination.schemas import PagedResponseSchema, T
+
 
 def paginate(
     query, response_schema: BaseModel, table: Table, page_params: PageParams, session: Session
     ) -> PagedResponseSchema[T]:
-    paginated_query = session.scalars(select(table).offset((page_params.page - 1) * page_params.size).limit(page_params.size)).all()
+    instances = session.scalars(query.offset((page_params.page - 1) * page_params.size).limit(page_params.size)).all()
+    total_amount = session.scalar(select(func.count()).select_from(table))
+    next_page_check = (total_amount - ((page_params.page - 1) * page_params.size)) > page_params.size
```

Comment:
Obsolete second query, if you want to query for n instances, query for n + 1 instead, then for next_page check use value from len(results) > n, for the results field in the pagination schema, pass results[:n] (skip that additional element queried earlier)